lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -66,32 +124,28 @@ optional<ConnectionState> PrusaLinkApi::accept(const RequestParser &parser) cons
         return get_only(SendStaticMemory("{\"printer\": {}}", http::ContentType::ApplicationJson, parser.can_keep_alive()));
     } else if (remove_prefix(suffix, "v1/files").has_value()) {
 
-        if (parser.method == Method::Put) {
-            static const auto prefix = "/api/v1/files/usb";
-            static const size_t prefix_len = strlen(prefix);
-            char fname[FILE_PATH_BUFFER_LEN + prefix_len];
-            if (!parser.uri_filename(fname, sizeof fname)) {
-                return StatusPage(Status::NotFound, parser.status_page_handling(), parser.accepts_json);
-            }
-
-            if (strlen(fname) <= prefix_len) {
-                return StatusPage(Status::Forbidden, parser.status_page_handling(), parser.accepts_json);
-            }
-
-            static const auto prefix_storage = "/api/v1/files";
-            static const size_t prefix_storage_len = strlen(prefix_storage);
-            char *storage_path = fname + prefix_storage_len;
-
-            if (strncmp(storage_path, "/usb/", 5) != 0) {
-                return StatusPage(Status::Forbidden, parser.status_page_handling(), parser.accepts_json);
-            }
-
+        static const auto prefix = "/api/v1/files";
+        static const size_t prefix_len = strlen(prefix);
+        char filename[FILE_PATH_BUFFER_LEN + prefix_len];
+        auto error = parse_file_url(parser, prefix_len, filename, FILE_PATH_BUFFER_LEN + prefix_len);
+        if (error != nullopt) {
+            return error;
+        }
+        switch (parser.method) {
+        case Method::Put: {
             GcodeUpload::PutParams putParams;
             putParams.print_after_upload = parser.print_after_upload;
-            strlcpy(putParams.filepath.data(), storage_path, sizeof(putParams.filepath));
+            strlcpy(putParams.filepath.data(), filename, sizeof(putParams.filepath));
             auto upload = GcodeUpload::start(parser, wui_uploaded_gcode, parser.accepts_json, std::move(putParams));
             return std::visit([](auto upload) -> ConnectionState { return std::move(upload); }, std::move(upload));
-        } else {
+        }
+        case Method::Get: {
+            return FileInfo(filename, parser.can_keep_alive(), parser.accepts_json, false);
+        }
+        case Method::Head: {
+            return FileInfo(filename, parser.can_keep_alive(), parser.accepts_json, false, true);
```

Comment:
I'd prefer to not put defaults in the parameters here; the FileInfo is called only from these two places.

Also, I'm thinking if passing the actual method (Head/Get) here would be more readable then true/false.